ouroboros-consensus/docs/report/chapters/future/genesis.tex

```diff
@@ -1499,3 +1500,178 @@ \subsection{Possible optimisations}
 to not being up to date would require careful thought.
 
 \end{itemize}
+
+\section{Suboptimal chain selection}
+
+The Ouroboros papers assume, by way of simplification, that nodes broadcast
+entire chains to each other. Every node collects all of these chains, and
+every slot selects the best chain they saw. In addition, the Ouroboros analysis
+assumes a constant $\Delta$, such that chains are submitted to all nodes in
+the network within $\Delta$ slots. A consequence of $\Delta > 1$ is that
+a newly forged block might not make it to the next slot leader in time; when
+this happens, that next slot leader will forge their block on a different tip
+instead, resulting in a short fork. As long as the chain propagation delay
+stays within $\Delta$, however, the Ouroboros analysis guarantees that
+consensus will nonetheless be achieved.
+
+In the implementation, each node follow the chains of its upstream peers as
+chains of \emph{headers}. These headers provide us with sufficient context to do
+chain selection and pick the best chain\footnote{Technically speaking this is a
+preliminary chain selection, because the chain we chose might contain invalid
+blocks. If this is the case, we will notice when we download the blocks and
+attempt to actually adopt the chain in the chain database. We will then
+disconnect from the peer that served us the offending chain, at which point a
+different chain might now be the preferred chain. This introduces a delay in
+selecting the optimal valid chain, but we accept this risk as sufficiently small
+that we do not pre-emptively download blocks from all chains. After all, there
+is little incentive for an attacker with the right to produce blocks to produce
+\emph{invalid} blocks.}. Under normal circumstances this is the \emph{only}
+chain we will download the blocks of and adopt, but occassionally we might want
+to download blocks from chains \emph{other} than the best chain, perhaps because
+downloading that best chain is taking a very long time (it might even be the
+case that the remote node is intentionally trying to stall us, as a form of
+denial of service attack). We will refer to this as \emph{suboptimal chain
+selection}; in this section we will discuss the conditions under which this is
+sound. Note that a simple argument that we might anyway not yet have seen the
+optimal chain does not hold water, due to the $\Delta$ guarantee.
+
+TODO\todo{TODO}: Should we think harder about pre-emptively download blocks?
+After all, a node giving us invalid blocks is \emph{also} introducing delays,
+which is precisely the kind of thing we are worried about here.
+One difference is that with an invalid block, we will disconnect; however,
+nothing is stopping a node from doing this again and again.
+I see two possible solutions: pre-emptively download blocks after all, or
+(more appealing) associate invalid blocks with signer keys?
```

Comment:
This is an interesting approach, but nothing is preventing an adversary of transferring the stake to a new fresh account and then bypass the banned signer key.